deployment/kubernetes.coffee

```diff
@@ -0,0 +1,99 @@
+fs            = require 'fs'
+{ isAllowed } = require './grouptoenvmapping'
+YAML          = require 'yamljs'
+
+generatePodDef = (podName) ->
+  podHeader =
+    apiVersion      : 'v1'
+    kind            : 'Pod'
+    metadata        :
+      name          : podName
+      namespace     : 'koding'
+    spec            :
+      restartPolicy : 'Never'
+      containers    : []
+
+  return podHeader
+
+generateContainerSection = (app, options = {}) ->
+
+  container =
+    name          : app
+    image         : options.kubernetes.image
+    workingDir    : '/opt/koding'
+    command       : options.kubernetes.command
+
+  container.ports = generatePortsSection options if options.ports
+  container.volumeMounts = options.kubernetes.mounts or []
+
+  return container
+
+generatePortsSection = (options) ->
+  return [ {
+    containerPort: parseInt(options.ports.incoming, 10),
+    hostPort: parseInt(options.ports.incoming, 10)
+  } ]
+
+generateVolumesSection = (KONFIG, options) ->
+  return [
+    { name: 'koding-working-tree', hostPath: { path: options.volumeDir } }
+    { name: 'assets', hostPath: { path: options.volumeDir + '/website' } }
+  ]
+
+module.exports.create = (KONFIG, options) ->
+  KONFIG.kubernetesConf = generatePodDef('backend')
+
+  for name, workerOptions of KONFIG.workers when workerOptions.kubernetes?.command?
+
+    unless isAllowed workerOptions.group, KONFIG.ebEnvName
+      continue
+
+    KONFIG.kubernetesConf.spec.containers.push generateContainerSection name, workerOptions
+
+  KONFIG.kubernetesConf.spec.volumes = generateVolumesSection KONFIG, options
```

Comment:
You should create the volumes spec from the KONFIG itself, not from your prior knowledge. This is the same reason that i commented earlier. 

Anyways please create an issue to refactor these volume creation steps and we can leave this as is for now.

Reply:
I opened an [issue](https://github.com/koding/koding/issues/11428) for it and will change it on the next iteration.